test/tools/test_from_function.py

```diff
@@ -1,8 +1,13 @@
 import pytest
 
-from haystack.tools.from_function import create_tool_from_function, _remove_title_from_schema, tool
 from haystack.tools.errors import SchemaGenerationError
-from typing import Annotated, Literal, Optional
+from haystack.tools.from_function import create_tool_from_function, _remove_title_from_schema, tool
+from typing import Literal, Optional
+
+try:
+    from typing import Annotated
+except ImportError:
+    from typing_extensions import Annotated
```

Comment:
`Annotated` has been added in Python 3.9, so we can avoid using `typing_extensions` here